dev/tools/localization/localizations_validator.dart

```diff
@@ -103,28 +101,28 @@ void validateLocalizations(
   Map<LocaleInfo, Map<String, String>> localeToResources,
   Map<LocaleInfo, Map<String, dynamic>> localeToAttributes,
 ) {
-  final Map<String, String> canonicalLocalizations = localeToResources[LocaleInfo.fromString('en')];
-  final Set<String> canonicalKeys = Set<String>.from(canonicalLocalizations.keys);
+  final Map<String, String>? canonicalLocalizations = localeToResources[LocaleInfo.fromString('en')];
+  final Set<String?> canonicalKeys = Set<String>.from(canonicalLocalizations!.keys);
   final StringBuffer errorMessages = StringBuffer();
   bool explainMissingKeys = false;
   for (final LocaleInfo locale in localeToResources.keys) {
-    final Map<String, String> resources = localeToResources[locale];
+    final Map<String, String>? resources = localeToResources[locale];
 
     // Whether `key` corresponds to one of the plural variations of a key with
     // the same prefix and suffix "Other".
     //
     // Many languages require only a subset of these variations, so we do not
     // require them so long as the "Other" variation exists.
     bool isPluralVariation(String key) {
-      final Match pluralMatch = kPluralRegexp.firstMatch(key);
+      final Match? pluralMatch = kPluralRegexp.firstMatch(key);
       if (pluralMatch == null)
         return false;
-      final String prefix = pluralMatch[1];
-      return resources.containsKey('${prefix}Other');
+      final String? prefix = pluralMatch[1];
+      return resources?.containsKey('${prefix}Other') ?? false;
     }
 
     final Set<String> keys = Set<String>.from(
-      resources.keys.where((String key) => !isPluralVariation(key))
+      resources!.keys.where((String key) => !isPluralVariation(key))
```

Comment:
If `resources` is non-null here, let's type it as such a few lines above.

Reply:
```
bool isPluralVariation(String key) {
      final Match? pluralMatch = kPluralRegexp.firstMatch(key);
      if (pluralMatch == null)
        return false;
      final String? prefix = pluralMatch[1];
      return resources?.containsKey('${prefix}Other') ?? false;
    }
```

resources can be null here.